tests/ui_tests/app/page/admin/page.py

```diff
@@ -489,3 +510,83 @@ def click_delete_button(self):
         self.wait_element_visible(DeleteDialog.body)
         self.find_and_click(DeleteDialog.yes)
         self.wait_element_hide(DeleteDialog.body)
+
+
+class AdminPoliciesPage(GeneralAdminPage):
+    """Admin Policy Page class"""
+
+    MENU_SUFFIX = 'policies'
+    MAIN_ELEMENTS = [
+        AdminIntroLocators.intro_title,
+        AdminIntroLocators.intro_text,
+        AdminGroupsLocators.create_group_btn,
+        AdminGroupsLocators.delete_btn,
+        CommonTable.header,
+    ]
+
+    def click_create_policy_btn(self):
+        self.find_and_click(AdminPoliciesLocators.create_policy_btn)
+        self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.block)
+
+    @allure.step('Create new policy')
+    def create_policy(
+        self, policy_name: str, description: Optional[str], role: str, users: Optional[str], groups: Optional[str]
+    ):
+        self.click_create_policy_btn()
+        self.send_text_to_element(AdminPoliciesLocators.AddPolicyPopup.FirstStep.name_input, policy_name)
+        if description:
+            self.send_text_to_element(AdminPoliciesLocators.AddPolicyPopup.FirstStep.description_input, description)
+        with allure.step(f"Select role {role} in popup"):
+            self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_select)
+            self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_item)
+            for available_role in self.find_elements(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_item):
+                if available_role.text == role:
+                    available_role.click()
+                    break
+        assert users or groups, "There are should be users or groups in the policy"
```

Comment:
I suggest moving this check to the first line of the function and make it not an assert:
```
if not (users or groups):
    raise ValueError("...")
```
The error can be anything else, just an example.

The motivation is following:
1. We validate input at the start and don't try to do things that won't lead us to success anyway.
2. Checking input as the assertion is the right way to do it in the program code, but assertion in our tests will lead to mark test as "having a bug", while not passing users or groups is definitely a test bug

Reply:
moved in the beginning

Reply:
Hm, I still disagree with using assert for what is a test creator error, not a bug. Can you please argue your decision to leave it as an assertion?

Reply:
yes. got it) changed